docs/technology_stack.adoc

```diff
@@ -7,35 +7,43 @@
 
-If you want to create a shared style that could be used by multiple components, then we recommend adding it as a mixin to
-*_src/assets/stylesheets/_planner-mixins.scss_*. The mixins are imported into every
-*_.component.scss_* file.
+== Development Pipeline
 
-You can then create a real class by doing something like:
+Following are the components or parts of the process followed in order to develop, build, test and deploy Planner. Most of these components are pretty standard, with limited project specific customizations/configurations applied.
 
-    .my-class {
-      @include my-class;
-    }
+Task Runner::
+We use Gulp for running predefined tasks. These tasks include, compiling/preprocessing the LESS/TypeScript sources into CSS/JS, building the distributable artifacts, running dev-server in watch mode and general house-keeping (install, reinstall, cleanup etc.).
 
-We use mixins to avoid polluting components with unnecessary style classes, and to avoid
-an explosion of shared files.
+Build System::
+We use both Gulp and webpack for building the source. While gulp takes care of the library builds, webpack is used for building the (minimal) `*_./runtime_*` to use the library, which makes Planner work as a standalone app. The build system works as intended for now, but can be expanded to use more sophisticated switches to build according to different needs (dev, tests, prod etc.).
 
-The *_src/assets/stylesheets/_* directory includes a *_shared_* directory. These are
-shared global styles which will be refactored into a shared library at some point.
-You can update these styles only if you are making a truly global style, and are going to
-synchronize your changes across all the various UI projects.
+Test Framework::
+Our test stack is built with Karma, Protractor and PhantomJS. PhantomJS gives the headless browser engine to run the app, which Protractor interacts with as a WebDriverJS wrapper and runs the tests according to Karma's configurations.
 
-== Gulp
-<TBD>
+Each `*_filename.ts_*` file is supposed to have unit tests along with them as a `*_filename.spec.ts_*` in the same parent directory. The functional tests reside inside `*_./runtime/tests/_*`. For more details about running the tests, please look into link:./testing.adoc[Testing Documentation].
+
+CI/CD::
+The CI/CD pipeline is made with Jenkins, as defined in the `*_./Jenkinsfile_*` at the root of the repo. It uses the content of `*_./deploy_*` (more specifically, `*_./deploy/release.groovy_*`) to orcastrate the CI/CD.
+
+Release Framework::
+We use semantic-release for versioning and publishing our Planner library to npm registry. It's fairly configuration less CLI task (`$ npm run semantic-release`) which takes the latest PR/merges and commit messages to account. It generates the new version number, publishes to npm registry, and then generates the release tag on GitHub in one simple process.
```

Comment:
It's> It is a, to account>into account
PR/merges>PR, merges
Global: substitute / with 'or'